ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -84,23 +84,92 @@ func (a *AviControllerInfra) VerifyAviControllerLicense() error {
 	return nil
 }
 
+func (a *AviControllerInfra) checkNSAnnotations() (error, string) {
+	cloudName := ""
+	nsName := utils.GetAKONamespace()
+	nsObj, err := a.cs.CoreV1().Namespaces().Get(context.TODO(), nsName, metav1.GetOptions{})
+	if err != nil {
+		utils.AviLog.Errorf("Failed to GET the %s namespace details due to the following error :%v", nsName, err.Error())
+		return err, cloudName
+	}
+	if nsObj.Annotations != nil {
+		cloudName = nsObj.Annotations["ako.vmware.com/wcp-cloud-name"]
+		if cloudName != "" {
+			utils.AviLog.Infof("Found cloud %s in ns annotation", cloudName)
+		}
+	}
+	return nil, cloudName
+}
+
+func (a *AviControllerInfra) checkVirtualService() (error, string) {
+	cloudName := ""
+	tenants := make(map[string]struct{})
+	err := lib.GetAllTenants(a.AviRestClient, tenants)
+	if err != nil {
+		utils.AviLog.Errorf("Error during GET tenants %v", err)
+		return err, cloudName
+	}
+	createdBy := lib.AKOPrefix + lib.GetClusterName()
```

Comment:
nit: would be better if we set AKO user during boot up, and then later reuse it as and when required.

Reply:
I had seen somewhere probably in main, where we initialise AKO user but this flow hits first so AKO user is not set at this time. I will let that part of code to set AKO user, not changing here.